File: NYCTaxiCabApp.Web/ClientApp/src/app/models/fare-summary.ts
import { Time } from '../models/time';
export interface IFareSummary{
    originAddress : string;
    destinationAddress: string        
    rideDate: Date;
    rideTime: Time;
    rideFare : number ;
    weekdayPeakSursarge : number ;
    nightSurcharge : number ;
    newYorkStateTax : number ;
    totalPrice: number ;
}

export class FareSummary implements IFareSummary{
    originAddress: string;
    destinationAddress: string;
    rideDate: Date;
    rideTime: Time;
    rideFare: number;
    weekdayPeakSursarge: number;
    nightSurcharge: number;
    newYorkStateTax: number;
    totalPrice: number;
    
}

File: NYCTaxiCabApp.Web/Models/FareSummary.cs
using NYCTaxiCabApp.Web.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace NYCTaxiCabApp.Web.Models
{
    public class FareSummary
    {
        public string OriginAddress { get; set; }
        public string DestinationAddress { get; set; }
        
        public DateTime RideDate { get; set; }
        public Time RideTime { get; set; }

        public decimal RideFare { get; set; }
        public decimal WeekdayPeakSursarge { get; set; }
        public decimal NightSurcharge { get; set; }
        public decimal NewYorkStateTax { get; set; }
        public decimal TotalPrice {
            get {
                    return this.RideFare + this.WeekdayPeakSursarge + this.NightSurcharge + 
                        this.NewYorkStateTax;
                }
        }

    }
}
